### ISSUE TITLE
defaults in .config are ignored

### ISSUE THREAD

**Issue by saraedum** [ffulm/firmware#4]
The defaults set in .config are ignored when changing the Target System through make menuconfig. In other words, if you do the following:

<pre>
cp firmware/.config .
make defconfig
make menuconfig (change Target System)
</pre>

then dnsmasq is again selected.

**Comment by saraedum** [comment]
(This is not a bug in the firmware, but in the README.)

**Comment by mwarning** [comment]
Yes, this is a sneaky problem. Any idea how to fix this? Maybe we should create a patch.

**Comment by mwarning** [comment]
Maybe something like this helps:

```
From 29a8abe25759e235e02ab70f2fba1693f65b7fb8 Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Tue, 27 Jan 2015 02:29:24 +0100
Subject: [PATCH 1/1] dnsmasq: make dhcpv6 support the default

---
 package/network/services/dnsmasq/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/network/services/dnsmasq/Makefile b/package/network/services/dnsmasq/Makefile
index 17af3c5..bc195eb 100644
--- a/package/network/services/dnsmasq/Makefile
+++ b/package/network/services/dnsmasq/Makefile
@@ -34,7 +34,8 @@ endef

 define Package/dnsmasq
 $(call Package/dnsmasq/Default)
-  VARIANT:=nodhcpv6
+  DEPENDS:=@IPV6 +kmod-ipv6
+  VARIANT:=dhcpv6
 endef

 define Package/dnsmasq-dhcpv6
-- 
2.1.4


```

**Comment by mwarning** [comment]
Let's see if someone knows the answer:
https://forum.openwrt.org/viewtopic.php?pid=263159

**Comment by mwarning** [comment]
ok, I've made a branch with fix. Needs to be tested:
https://github.com/ffulm/firmware/tree/dnsmasq

**Comment by mwarning** [comment]
I have submitted a patch that has a different approach: https://dev.openwrt.org/ticket/18850
It also needs to be tested.

**Comment by mwarning** [comment]
Fixed in 5182decfbeccd390b69c494a75748bb2036090b8 for now.
